tutorials/tutorial_uploads.md

```diff
@@ -27,7 +28,169 @@ The Procore API provides the following endpoints for creating Uploads at the com
+
+### Step 2 - Create a Segmented Upload
+
+To help illustrate the workflow for this example we use the Create Upload endpoint to create a new segmented upload at the company level.
+A sample POST request to the [Create Upload](https://developers.procore.com/reference/rest/v1/company-uploads) endpoint would take the following format:
+
+- Request Method: `POST`
+- Request URL: `https://api.procore.com/rest/v1.0/companies/{company_id}/uploads`
+- Request Body: An array of file segment information with each segment containing `size` (in bytes), `sha256` (sha-256 hash), `md5` (optional md5 checksum).
+
+**Example Request Body**
+
+```
+{
+    "segments": [
+        {
+        "size": 128341,
+        "sha256": "1ba7385b7a1bb841178734fa8fb5924d3082364c6d693cc5be3026a468a56220",
+        "md5": "514461e89149642ca1ca94aa363289ff"
+        }
+    ]
+}
+```
+
+**Response Body**
+
+The response will include the `UUID` for the file, and the specific information required to upload each file segment to the third-party storage service.
+
+**UUID** - the unique identifier for the binary file
+
+**Segments** - array of file segment information containing:
+* **MD5** - the MD5 checksum for the file segment
+* **SHA256** - the SHA-256 hash of the file segment
+* **Size** - the file segment size in bytes
+* **URL** - the upload host url of the third party storage service
+* **Headers** - the required request headers to upload the segment
+
+**Status** - the current status of the direct file upload
+
+Example:
+
+```
+{
+    "uuid": "01FPTX6G2X9MWKSNG8V232VZZZ",
+    "segments": [
+        {
+            "md5": "514461e89149642ca1ca94aa363289ff",
+            "sha256": "1ba7385b7a1bb841178734fa8fb5924d3082364c6d693cc5be3026a468a56220",
+            "size": 128341,
+            "url": "https://s3.amazonaws.com/pro-core.com/companies/xyz/...",
+            "headers": {
+                "x-amz-content-sha256": "1ba7385b7a1bb841178734fa8fb5924d3082364c6d693cc5be3026a468a56220",
+                "content-length": "128341",
+                "content-md5": "UURh6JFJZCyhypSqNjKJ/w=="
+            }
+        }
+    ],
+    "status": "ready"
+}
+```
+
+### Step 3 - Upload File Segment(s) to Storage Service
+
+After successfully creating a new upload, the next step is to upload all the file segments to the third-party storage provider.
+The response from Create Upload in the previous step includes the required URL and header values for each segment that must be sent to the storage provider with each upload request.
+An eTag is returned in the response for each segment upload. We'll need to keep each eTag value for the next step.
+
+**Request**
+
+- Request Method: `PUT`
+- URL: Segment url value from the create upload response
+- AUTH: None
+- Request Headers: Segment header values from the create upload response
+- Request Body: File binary
```

Comment:
This is confusing to me. How is the file supposed to be included in the request body here? Is there a key that is included? Am I uploading the actual hexdump of the file that I used to generate the sha256 hash? I don't know enough about segmented uploads to know what this means. I think an example would be helpful here. I tried to do this and get an error saying "The provided 'x-amz-content-sha256' header does not match what was computed." so I must have done something wrong.

Reply:
Yeah, Steps 3 and 4 need some additional work. @njaitken-pc I'll work with you to improve these sections.

Reply:
It is plain old http binary body, no encoding, nothing special.